src/ts/components/timelinemarkershandler.ts

```diff
@@ -0,0 +1,209 @@
+
+  public configure(player: PlayerAPI, uimanager: UIInstanceManager) {
+    this.player = player;
+    this.uimanager = uimanager;
+    this.configureMarkers();
+  }
+
+  private configureMarkers(): void {
+    // Remove markers when unloaded
+    this.player.on(this.player.exports.PlayerEvent.SourceUnloaded, () => this.clearMarkers());
+    // Update markers when the size of the seekbar changes
+    this.player.on(this.player.exports.PlayerEvent.PlayerResized, () => this.updateMarkersDOM());
+
+    this.player.on(this.player.exports.PlayerEvent.SourceLoaded, () => {
+      if (this.player.isLive()) {
+        // Update marker position as timeshift range changes
+        this.player.on(this.player.exports.PlayerEvent.TimeChanged, () => this.updateMarkers());
+        // Update marker postion when paused as timeshift range changes
+        this.configureLivePausedTimeshiftUpdater(() => this.updateMarkers());
+      }
+    });
+    this.uimanager.getConfig().events.onUpdated.subscribe(() => this.updateMarkers());
+    this.uimanager.onRelease.subscribe(() => this.uimanager.getConfig().events.onUpdated.unsubscribe(() => this.updateMarkers()));
+
+    // Init markers at startup
+    this.updateMarkers();
+  }
+
+  public getMarkerAtPosition(percentage: number): SeekBarMarker | null {
+    const snappingRange = this.config.snappingRange;
+
+    if (this.timelineMarkers.length > 0) {
+      this.timelineMarkers.forEach(marker => {
+        // Handle interval markers
+        if (marker.duration > 0
+          && percentage >= marker.position - snappingRange
+          && percentage <= marker.position + marker.duration + snappingRange) {
+          return marker;
+        }
+        // Handle position markers
+        else if (percentage >= marker.position - snappingRange
+          && percentage <= marker.position + snappingRange) {
+          return marker;
+        }
+      });
+    }
+
+    return null;
+  }
+
+  private clearMarkers(): void {
+    this.timelineMarkers = [];
+    this.markersContainer.empty();
+  }
+
+  private removeMarker(marker: TimelineMarker): void {
+    this.uimanager.getConfig().metadata.markers = this.uimanager.getConfig().metadata.markers.filter(_marker => marker !== _marker);
+  }
+
+  private updateMarkers(): void {
+    this.clearMarkers();
+
+    if (!shouldProcessMarkers(this.player, this.uimanager)) {
+      return;
+    }
+
+    this.uimanager.getConfig().metadata.markers.forEach(marker => {
+      const { markerPosition, markerDuration } = getMarkerPositions(this.player, marker);
+
+      if (shouldRemoveMarker(markerPosition, markerDuration)) {
+        this.removeMarker(marker);
+      } else {
+        this.timelineMarkers.push({ marker, position: markerPosition, duration: markerDuration });
+      }
+    });
+
+    // Populate the timeline with the markers
+    this.updateMarkersDOM();
+  }
+
+  private updateMarkersDOM(): void {
+    // TODO: For live streams we are removing all markers, and re-adding them.
```

Comment:
Can you add a jira task for the follow up?

Reply:
Will communicate out if needed.